node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/Rebalancer.java

```diff
@@ -90,20 +107,50 @@ private Move findBestMove(NodeList allNodes) {
                 double skewReductionAtToHost = skewReductionByAdding(node, toHost, capacity);
                 double netSkewReduction = skewReductionAtFromHost + skewReductionAtToHost;
                 if (netSkewReduction > bestMove.netSkewReduction)
-                    bestMove = new Move(node, netSkewReduction);
+                    bestMove = new Move(node, toHost, netSkewReduction);
             }
         }
         return bestMove;
     }
 
-    private void markWantToRetire(Node node) {
+    private boolean markWantToRetire(Node node, boolean wantToRetire) {
         try (Mutex lock = nodeRepository().lock(node)) {
             Optional<Node> nodeToMove = nodeRepository().getNode(node.hostname());
-            if (nodeToMove.isEmpty()) return;
-            if (nodeToMove.get().state() != Node.State.active) return;
+            if (nodeToMove.isEmpty()) return false;
+            if (nodeToMove.get().state() != Node.State.active) return false;
+
+            if (node.status().wantToRetire() != wantToRetire)
```

Comment:
If they are equal, we should return `false` or at least make sure not to "undo" this in `finally` if we fail to achieve the wanted result.

Reply:
If it's already set to want to retire it's fine that we do it.
However, as you point out it is not fine to undo it if somebody else sets it and we failed (regardless of when it happens).

PTAL - I'm going to run out of "try" clauses soon.

Reply:
PTAL